dot-object/index.d.ts

```diff
@@ -109,7 +109,7 @@ declare namespace DotObject {
          * @param {Object} obj
          * @param {Boolean} remove
          */
-        pick(path: string, obj: any, remove?: boolean): void;
+        pick(path: string, obj: any, remove?: boolean): any;
```

Comment:
- tested this in my production code, and TypeScript was now happy.
- did this according to my experience with using the function and the documentation as described in https://github.com/rhalff/dot-object#pickremove-a-value-using-dot-notation